nni/retiarii/graph.py

```diff
@@ -309,7 +309,7 @@ def add_node(self, name: str, operation: Operation) -> 'Node': ...
     @overload
     def add_node(self, name: str, type_name: str, parameters: Dict[str, Any] = {}) -> 'Node': ...
 
-    def add_node(self, name, operation_or_type, parameters={}):
+    def add_node(self, name, operation_or_type, parameters=None):
```

Comment:
Why change the behavior of this API?

Reply:
From my knowledge, it is not a good practice to use `{}` as default values, here is a reference 
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Not sure whether there is any other special reason, but I believe it worth a refactor to change all default `{}` and `[]` to `None`

Reply:
> Why change the behavior of this API?

And this will cause parameters of the nodes to share the same dict object (the initial `{}`). When one of the parameters is changed, all of the others will also be changed at the same time. 

This results in some strange bugs.

Reply:
I'm okay with this change if you insist. But you should put a `if parameters is None: parameters = {}` in the body of this function.

Reply:
If parameters is immutable in the function body, actually putting `parameters={}` in the arguments works, except that IDE will complain.

Reply:
> I'm okay with this change if you insist. But you should put a if parameters is None: parameters = {} in the body of this function.

Yes. I have added that to `Opeartion::__init__()` and `Operation::__new__()`.

> If parameters is immutable in the function body, actually putting parameters={} in the arguments works, except that IDE will complain.

But if you assign it to the attribute of the object, when next time you modify that attribute, the other objects will also be affected. (Actually this is why it incurs some strange bugs and it costs me a night to debug out 😂